src/mpags-cipher.cpp

```diff
@@ -1,60 +1,54 @@
 #include <cctype>
 #include <iostream>
+#include <fstream>
 #include <string>
 #include <vector>
 
-int main(int argc, char* argv[])
+#include "TransformChar.hpp"
+#include "ProcessCommandLine.hpp"
+
+std::string transformChar(const char in);
+
+int processCommandLine(
+    const std::vector<std::string>& args,
+    bool& helpRequested,
+    bool& versionRequested,
+    bool& encrpt,
+    unsigned long& key,
+    size_t& cipherTrigger, // this number detects whether -d or -e flag added
+    std::string& inputFileName,
+    std::string& outputFileName
+);
+
+std::string runCaesarCipher( 
+    const std::string& inputText, 
+    const size_t key,
+    const bool encrpt
+);
```

Comment:
Need to include also `RunCaesarCipher.hpp`, then you can drop these explicit function declarations here since they are already in the files you include.

```suggestion
#include "TransformChar.hpp"
#include "ProcessCommandLine.hpp"
#include "RunCaesarCipher.hpp"
```